### PR TITLE
Be more precise when setting a property with invalid type

### PR DESCRIPTION
**Pull request by fengalin**
When setting an `Object` property with a valid name but an invalid type, `"Invalid property name"` is returned. This modification allows returning `"Invalid property type"` in this case.

Changes return type for `Object::has_property`.

### CODE REVIEW

File: src/object.rs [hunk 1]
```diff
@@ -628,6 +629,20 @@ impl<T: IsA<Object> + SetValue> ObjectExt for T {
         }
     }
 
+    fn has_property_with_type<'a, N: Into<&'a str>>(&self, property_name: N, type_: Type) -> Result<(), BoolError> {
+        let property_name = property_name.into();
+        let ptype = self.get_property_type(property_name);
+
+        match ptype {
+            None => Err(BoolError("Invalid property name")),
+            Some(ptype) => if ptype == type_ {
```
Comment (GuillaumeGomez) [inline src/object.rs]: Please put this one in a block, it's not really beautiful like this (even though it's perfectly correct).

**Comment by GuillaumeGomez** [comment]
This is awesome! Just a nit and it's good for me.

cc @EPashkin @sdroege

File: src/object.rs (hunk 1 above)
Comment (fengalin) [inline src/object.rs]: Yeah, I agree! I'm just not too sure about what the custom is. Would that be better:
``` rust
        if let Some(ptype) = ptype {
            if ptype == type_ {
                Ok(())
            } else {
                Err(BoolError("Invalid property type"))
            }
        } else {
            Err(BoolError("Invalid property name"))
        }
```

Comment (GuillaumeGomez) [inline src/object.rs]: Not worst at least so let's go for this solution! :)

Comment (fengalin) [inline src/object.rs]: Ça marche ! ;)

Comment (GuillaumeGomez) [inline src/object.rs]: Keep it in english so other people can follow as well please (même si je pense que tout le monde aura compris ce que t'as dit / even though I think everyone understood what you wrote). :)

**Comment by sdroege** [comment]
It's fine to change API IMHO, these functions are all new and in no release yet

**Comment by sdroege** [comment]
I.e., please just change the existing function :)

**Comment by fengalin** [comment]
All right, I'll do that!

**Comment by fengalin** [comment]
By the way, any suggestion on the "ugly part"?

**Comment by sdroege** [comment]
What ugly part?

**Comment by fengalin** [comment]
It used to be:
``` rust
        match ptype {
            None => Err(BoolError("Invalid property name")),
            Some(ptype) => if ptype == type_ {
                    Ok(())
                } else {
                    Err(BoolError("Invalid property type"))
                },
        }
```
I must admit, I was not too proud of it. Then upon @GuillaumeGomez request, I proposed this:
``` rust
        if let Some(ptype) = ptype {
            if ptype == type_ {
                Ok(())
            } else {
                Err(BoolError("Invalid property type"))
            }
        } else {
            Err(BoolError("Invalid property name"))
        }
```
But I'm still not convinced.

**Comment by fengalin** [comment]
Never mind... I got it known.

**Comment by sdroege** [comment]
Looks good to me

**Comment by EPashkin** [comment]
LGFM, only commit title better changed.

**Comment by fengalin** [comment]
Do you want me to squash the commits and keep the first commit message only?

**Comment by GuillaumeGomez** [comment]
Yes please!

**Comment by fengalin** [comment]
Not sure if you want me to do something else on this one.

@EPashkin: shall I enhance "Be more precise when setting a property with invalid type"?

**Comment by EPashkin** [comment]
Yes, after squash all good for me. Thanks.

**Comment by GuillaumeGomez** [comment]
Well, he already squashed so let's merge. Thanks!

**Pull request merged**